### PR TITLE
Refactor moisture handling with scheme-dependent prognostic naming

### PR DESCRIPTION
**Pull request by glwagner**
## Summary

- Replace hardcoded `:ρqᵗ` prognostic key with scheme-dependent names: `:ρqᵛ` (non-equilibrium/Nothing) and `:ρqᵉ` (equilibrium moisture density --- often vapor, cloud liquid, and cloud ice)
- Rename `model.vapor_density` → `model.moisture_density`; remove `model.specific_moisture` field (replaced by `specific_prognostic_moisture(model)` function accessor)
- Non-equilibrium microphysics schemes now store true vapor `qᵛ` as the prognostic variable instead of total moisture `qᵗ`
- Add `grid_moisture_fractions` (reads vapor directly from fields for grid models) vs `moisture_fractions` (subtracts condensate from total moisture for parcel models)

### New public API
- `moisture_prognostic_name(microphysics)` → `:ρqᵛ` or `:ρqᵉ`
- `moisture_specific_name(microphysics)` → `:qᵛ` or `:qᵉ`
- `specific_prognostic_moisture(model)` → specific moisture field from microphysical fields

### Breaking changes
- `model.vapor_density` → `model.moisture_density`
- `model.specific_moisture` removed (use `specific_prognostic_moisture(model)`)
- `vapor_mass_fraction` removed (use `specific_humidity`)
- Boundary condition keys: `:ρqᵗ` → `:ρqᵛ` (or `:ρqᵉ` for SA)

## Test plan
- [x] Full test suite passes (1521/1521)
- [x] Run BOMEX example (SA path with `:ρqᵉ`)
- [x] Run cloudy_thermal_bubble example (non-eq path with `:ρqᵛ`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: examples/prescribed_sea_surface_temperature.jl [hunk 1]
```diff
@@ -326,7 +326,7 @@ Jᵛ = BoundaryConditionOperation(ρqᵗ, :bottom, model)
 # helping monitor the simulation's progress and detect any numerical issues.
 
 function progress(sim)
-    qᵗ = sim.model.specific_moisture
+    qᵗ = specific_humidity(sim.model)
```
Comment (kaiyuan-cheng) [inline examples/prescribed_sea_surface_temperature.jl]: same here

File: examples/prescribed_sea_surface_temperature.jl [hunk 2]
```diff
@@ -361,7 +361,7 @@ add_callback!(simulation, progress, IterationInterval(100))
 # The JLD2 format provides efficient storage with full Julia type preservation.
 
 output_filename = "prescribed_sea_surface_temperature_convection.jld2"
-qᵗ = model.specific_moisture
+qᵗ = specific_humidity(model)
```
Comment (kaiyuan-cheng) [inline examples/prescribed_sea_surface_temperature.jl]: same here

File: src/AtmosphereModels/atmosphere_model_buoyancy.jl [hunk 3]
```diff
@@ -33,9 +33,10 @@ OceanTurbulenceClosures.buoyancy_force(model::AtmosphereModel) =
 # TODO: make this microphysics-aware, and also saturation/condensate-aware
 function OceanTurbulenceClosures.buoyancy_tracers(model::AtmosphereModel)
     # Diagnostic fields for buoyancy gradient calculation
-    buoyancy_tracers = (; T = model.temperature, qᵗ = model.specific_moisture)
+    buoyancy_tracers = (; T = model.temperature, qᵛ = specific_humidity(model))
```
Comment (kaiyuan-cheng) [inline src/AtmosphereModels/atmosphere_model_buoyancy.jl]: I guess that we still need qᵗ here.

File: src/BoundaryConditions/bulk_scalar_fluxes.jl [hunk 4]
```diff
@@ -152,8 +152,8 @@ Base.summary(bf::BulkVaporFluxFunction) =
     ρ₀ = surface_density(p₀, T₀, constants)
     qᵛ₀ = saturation_specific_humidity(T₀, ρ₀, constants, surface)
 
-    qᵗ = @inbounds fields.qᵗ[i, j, 1]
-    Δq = qᵗ - qᵛ₀ # neglecting condensate
+    qᵛ = @inbounds fields.qᵛ[i, j, 1]
+    Δq = qᵛ - qᵛ₀
```
Comment (kaiyuan-cheng) [inline src/BoundaryConditions/bulk_scalar_fluxes.jl]: This becomes consistent 😄

File: src/StaticEnergyFormulations/static_energy_tendency.jl [hunk 5]
```diff
@@ -201,12 +201,12 @@ end
...
         pᵣ = dynamics_pressure(dynamics)[i, j, k]
         ρᵣ = dynamics_density(dynamics)[i, j, k]
-        qᵗ = specific_moisture[i, j, k]
+        qᵛ = specific_moisture[i, j, k]
         T = temperature_field[i, j, k]
     end
 
     # Get moisture fractions (vapor only for unsaturated air)
-    q = grid_moisture_fractions(i, j, k, grid, microphysics, ρᵣ, qᵗ, microphysical_fields)
+    q = grid_moisture_fractions(i, j, k, grid, microphysics, ρᵣ, qᵛ, microphysical_fields)
```
Comment (kaiyuan-cheng) [inline src/StaticEnergyFormulations/static_energy_tendency.jl]:
```suggestion
    q = grid_moisture_fractions(i, j, k, grid, microphysics, ρᵣ, qₘ, microphysical_fields)
```

File: src/PotentialTemperatureFormulations/potential_temperature_tendency.jl [hunk 6]
```diff
@@ -231,12 +231,12 @@ end
     @inbounds begin
         pᵣ = dynamics_pressure(dynamics)[i, j, k]
         ρᵣ = dynamics_density(dynamics)[i, j, k]
-        qᵗ = specific_moisture[i, j, k]
+        qᵛ = specific_moisture[i, j, k]
```
Comment (kaiyuan-cheng) [inline src/PotentialTemperatureFormulations/potential_temperature_tendency.jl]:
```suggestion
        qₘ = specific_moisture[i, j, k]
```

File: src/PotentialTemperatureFormulations/potential_temperature_tendency.jl [hunk 7]
```diff
@@ -160,14 +160,14 @@ end
...
```
Comment (kaiyuan-cheng) [inline src/PotentialTemperatureFormulations/potential_temperature_tendency.jl]:
```suggestion
        qₘ = specific_moisture[i, j, k]
```

File: src/PotentialTemperatureFormulations/potential_temperature_tendency.jl [hunk 8]
```diff
@@ -160,14 +160,14 @@ end
...
         e = specific_energy[i, j, k]
     end
 
     z = znode(i, j, k, grid, c, c, c)
-    q = grid_moisture_fractions(i, j, k, grid, microphysics, ρᵣ, qᵗ, microphysical_fields)
+    q = grid_moisture_fractions(i, j, k, grid, microphysics, ρᵣ, qᵛ, microphysical_fields)
     𝒰e₀ = StaticEnergyState(e, q, z, pᵣ)
-    𝒰e₁ = maybe_adjust_thermodynamic_state(𝒰e₀, microphysics, qᵗ, constants)
+    𝒰e₁ = maybe_adjust_thermodynamic_state(𝒰e₀, microphysics, qᵛ, constants)
```
Comment (kaiyuan-cheng) [inline src/PotentialTemperatureFormulations/potential_temperature_tendency.jl]:
```suggestion
    𝒰e₁ = maybe_adjust_thermodynamic_state(𝒰e₀, microphysics, qₘ, constants)
```

File: src/AtmosphereModels/atmosphere_model_buoyancy.jl (hunk 3 above)
Comment (glwagner) [reply src/AtmosphereModels/atmosphere_model_buoyancy.jl]: @kaiyuan-cheng are you sure? For buoyancy we need to compute density, which depends on q^v rather than q^t, right?

File: src/AtmosphereModels/atmosphere_model_buoyancy.jl [hunk 9]
```diff
@@ -44,16 +45,16 @@ end
...
     g = b.thermodynamic_constants.gravitational_acceleration
-    ∂z_ϑ = ∂zᶜᶜᶠ(i, j, k, grid, virtual_potential_temperature, b.thermodynamic_constants, b.dynamics, tracers.T, tracers.qᵗ)
-    ϑ = virtual_potential_temperature(i, j, k, grid, b.thermodynamic_constants, b.dynamics, tracers.T, tracers.qᵗ)
+    ∂z_ϑ = ∂zᶜᶜᶠ(i, j, k, grid, virtual_potential_temperature, b.thermodynamic_constants, b.dynamics, tracers.T, tracers.qᵛ)
+    ϑ = virtual_potential_temperature(i, j, k, grid, b.thermodynamic_constants, b.dynamics, tracers.T, tracers.qᵛ)
     return g * ∂z_ϑ / ϑ
 end
 
-@inline function virtual_potential_temperature(i, j, k, grid, constants, dynamics, T, qᵗ)
+@inline function virtual_potential_temperature(i, j, k, grid, constants, dynamics, T, qᵛ)
```
[the remark this answers is not in the document]
Comment (glwagner) [reply src/AtmosphereModels/atmosphere_model_buoyancy.jl]: Good point — this should account for condensate loading. There is already a TODO on line 33 for making this microphysics-aware and condensate-aware. For now it only uses vapor (same as before this PR), but we should fix it in a follow-up.

File: src/Microphysics/microphysics_diagnostics.jl [hunk 10]
```diff
@@ -141,14 +141,15 @@ function (d::AdjustmentRH)(i, j, k, grid)
         pᵣ = d.reference_state.pressure[i, j, k]
         ρᵣ = d.reference_state.density[i, j, k]
         T = d.temperature[i, j, k]
-        qᵗ = d.specific_moisture[i, j, k]
+        # qᵛᵉ: vapor (non-equilibrium) or equilibrium moisture (saturation adjustment)
+        qᵛᵉ = d.specific_moisture[i, j, k]
```
Comment (glwagner) [inline src/Microphysics/microphysics_diagnostics.jl]: `d.prognostic_specific_moisture` ?
